client/trino-client/src/main/java/io/trino/client/QueryDataJacksonModule.java

```diff
@@ -127,6 +127,13 @@ public void serialize(Segment value, JsonGenerator gen, SerializerProvider provi
                     .serializeWithType(value, gen, provider, segmentSerializer(provider));
         }
 
+        @Override
+        public void serializeWithType(Segment value, JsonGenerator gen, SerializerProvider provider, TypeSerializer typeSer)
```

Comment:
why is typeSer not used?

Reply:
It is used if you want to do a custom representation using JsonGenerator

Reply:
This will be an instance of `AsPropertyTypeSerializer`

Reply:
I've used the type explicitly